libsplinter/src/circuit/store.rs

```diff
@@ -36,11 +36,46 @@ impl CircuitFilter {
 }
 
 pub trait CircuitStore: Send + Sync + Clone {
-    fn circuits(&self) -> Result<BTreeMap<String, Circuit>, CircuitStoreError>;
+    /// Return an iterator over the circuits in this store.
+    ///
+    /// A circuit filter may optionally provided, to reduces the results.
+    fn circuits(&self, filter: Option<CircuitFilter>) -> Result<Circuits, CircuitStoreError>;
```

Comment:
circuit/circuits Circuit/Circuits are very closely named, maybe list_circuits()->CircuitList?

Reply:
Should we just return Iterator<Item = Circuit> here so we avoid exposing the implementation completely?

Reply:
Unfortunately, no.  I tried to go down that route with event iteration, and the compiler did not like it due to the async layer at the REST API level. We may be able to change that when we more cleanly separate the rest API business logic layer from the underlying http server library. 

It's not a list, though, so I don't think CircuitList isn't the right name.  This is also a pretty common pattern in the standard library:  See `String::chars() -> Chars` as a good example. I could probably be convinced to call it `CircuitIter`, though.

Reply:
CircuitIter seems fine

Reply:
Will add an new PR for this.